Automattic/wp-calypso
# Add GitHub Issue Template block to o2-blocks

#### Changes proposed in this Pull Request


- Add a P2 block that looks like the GitHub shortcode and allows creating Github issues based on a pre-defined template. Useful to store Github issue templates internally in field guides pages. After being saved, the block saves a URL that has all the needed information to create the issue with pre-filled data in a specific user or org repository;
- Add a plugin `index.php` so that it can be installed as a plugin from local WordPress instances (when developing locally, see the `Testing Instructions` section below for more information);
- Update README to add instructions on how to develop `o2-blocks` locally


#### Testing instructions

1. Clone this repo and `cd` to `<repo_dir>/apps/o2-blocks`;
2. Run `yarn` and then `yarn build`;
3. Install  `wp-env` (if you don't yet have it) and from the same dir state above, run `wp-env start`;
4. Open a web browser and go to `http://localhost:8889/wp-admin/post-new.php`;
5. Click the `+` button on the page to bring the block picker, then search for `GitHub Issue Template Button`, once found, click it in order to add it to the page
6. Click the button and click the gear icon on the top-right corner of the page to bring the Settings inspector;
7. Fill the data, check the screencast below for an example
8. Preview the page and click the button

9. You should be taken to the GitHub new Issue form with all the data you entered on step `7` pre-filled. Yay, an issue template!

#### Screenshots
| Editing | View in the editor | Validation Error | View in the frontend | After clicking it in the frontend |
| --- | --- | --- | --- | --- |
| ![newghblockedit](https://user-images.githubusercontent.com/81248/88993563-f16e4e00-d2ab-11ea-9541-68995d8f4925.png) | ![newghblockeditorview](https://user-images.githubusercontent.com/81248/88993576-f7fcc580-d2ab-11ea-8369-f1b56311f3f9.png) | ![blockwarnign](https://user-images.githubusercontent.com/81248/89068792-95e9a200-d337-11ea-8d15-91bb1914700e.png) | ![newghblockfrontendview](https://user-images.githubusercontent.com/81248/89068747-7fdbe180-d337-11ea-90d5-eb484a695403.png) | ![newghblockrepoissuecreate](https://user-images.githubusercontent.com/81248/88993607-08ad3b80-d2ac-11ea-92bd-81f52f54d282.png) |



#### TODO
- [ ] Create the wpcom diff with the release + adding the enqueue code for the "view.css" to the a8c blocks plugin loader PHP script in WPCOM
- [ ] Once approved, pair with someone with WPCOM SVN commit access to deploy it.
